### PR TITLE
feat(next): add sentry to payments-next

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- Be able to use Sentry in Next.js on both the client and the server

## This pull request

- Add Sentry to payments-next using the installation wizard

## Issue that this pull request solves

Closes: #FXA-9998

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: apps/payments/next/app/sentry-example/page.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
+'use client';
```
Comment (StaberindeZA) [inline apps/payments/next/app/sentry-example/page.tsx:1]: To be deleted before merging.

File: apps/payments/next/app/global-error.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (StaberindeZA) [inline apps/payments/next/app/global-error.tsx:1]: To be deleted before merging.

File: package.json [hunk 3]
```diff
@@ -276,7 +277,8 @@
     "tap/typescript": "^4.5.2",
     "terser:>4.0.0 <5": ">=4.8.1",
     "terser:>5 <6": ">=5.14.2",
-    "underscore": ">=1.13.2"
+    "underscore": ">=1.13.2",
+    "@sentry/types": "^7.113.0"
```
Comment (StaberindeZA) [inline package.json:281]: This should be removed once #17505 is merged.

File: tsconfig.base.json [hunk 4]
```diff
@@ -61,6 +61,7 @@
       "@fxa/shared/pem-jwk": ["libs/shared/pem-jwk/src/index.ts"],
       "@fxa/shared/react": ["libs/shared/react/src/index.ts"],
       "@fxa/shared/sentry": ["libs/shared/sentry/src/index.ts"],
+      "@fxa/shared/sentry/client": ["libs/shared/sentry/src/client.ts"],
```
Comment (StaberindeZA) [inline tsconfig.base.json:64]: Some modules in `shared-sentry` require modules only available in a `node` environment, such as `fs`, which are not available on the client (browser). Therefore a separate `client` barrel file is required which would not include these modules.

**Review by julianpoy — APPROVED** [review]

**Pull request merged**